src/plugins/record.ts

```diff
@@ -148,6 +148,8 @@ class RecordPlugin extends BasePlugin<RecordPluginEvents, RecordPluginOptions> {
     if (this.isRecording()) {
       this.mediaRecorder?.stop()
       this.stream?.getTracks().forEach((track) => track.stop())
+      this.stream = null
+      this.mediaRecorder = null
```

Comment:
Actually, this whole thing is incorrect. When recording is stopped, it shouldn't stop the stream. The stream should be stopped only when the mic is stopped (the `stopMic` method is called).

The idea is that you can monitor the microphone across recordings. You can stop the mic explicitely from your app.

Reply:
That's because `this.mediaRecorder` is cached and is never re-created. So it keeps the old stream.

Reply:
We need to recreate the MediaRecorder object when the mic stream changes. https://github.com/katspaugh/wavesurfer.js/blob/main/src/plugins/record.ts#L105-L111

Reply:
We should then set `this.mediaRecorder = null` inside `stopMic` because then, in the `startRecording` function, `this.mediaRecorder` will be `null` and 
```
      new MediaRecorder(stream, {
        mimeType: this.options.mimeType || findSupportedMimeType(),
        audioBitsPerSecond: this.options.audioBitsPerSecond,
      })
```

gets executed thereby recreating the MediaRecorder object when the mic stream changes.

Reply:
Yes, sounds right

Reply:
@katspaugh made the change.